### PR TITLE
test: upgrade pytest and friends, cleanup configuration

### PR DESCRIPTION
**Pull request by joshuarli**
After reading a bunch of changelogs, this upgrades pytest and friends to a good set of versions for increased harmony with the upcoming Django 2.2 and Python 3.8.

Not too well-versed in pytest, but most of the few new features over these versions didn't seem really interesting to me. Mostly bugfixes and improvements.

After pytest 6+ we're able to consolidate configuration into pyproject.toml which is preferable, I went ahead and did that and also cleaned up some old configuration that doesn't apply / isn't useful anymore. `--strict-markers` is good. I also tried to use the new `--import-mode=importlib` but that became a TODO.

I'll follow up with fixing all the warnings and then forbidding warnings with some filters, there's a lot. (There has been a lot even before this.)

### CODE REVIEW

**Comment by joshuarli** [comment]
Mmm, output looks very different than what I'm used to now, will have to revisit configuration.

**Comment by joshuarli** [comment]
Oh, this is what the old `-p no:warnings` did. I assumed I could remove it since it wasn't listed in the plugin list in the test header. It seems to be something special.

That's confusing, but looks like `--disable-warnings` does the same thing, so I'll put that there instead for now. It's extremely verbose, so I have to fix the warnings that should be fixed, remove that since ideally we wanna see warnings, and add filters for stuff like Python 3.6 and Django 20 and 21 until those are fixed.

**Comment by joshuarli** [comment]
Oh huh with `-p no:warnings` there's no warning count at the end either compared to `--disable-warnings`. Guess I'll go with the former then actually.

File: setup.cfg [hunk 1]
```diff
@@ -1,15 +1,3 @@
-[tool:pytest]
-python_files = test*.py
-# Note: When updating pytest tweaks, make sure to update getsentry
-# When updating the traceback format, make sure to update .github/pytest.json
-addopts = --tb=short -p no:doctest -p no:warnings
-norecursedirs = bin dist docs htmlcov script hooks node_modules .* {args}
-looponfailroots = src tests
```
Comment (billyvg) [inline setup.cfg:7]: what about these options?

Comment (joshuarli) [reply setup.cfg:7]: norecursedirs: Our pytest invocations already specify filepaths. Usually you specify these if you're just invoking `pytest` straight from gitroot.

looponfailroots: undocumented and as far as I can tell, isn't useful

Comment (joshuarli) [reply setup.cfg:7]: edit: looponfailroots is part of pytest-xdist

Comment (joshuarli) [reply setup.cfg:7]: I added it back.

Comment (wedamija) [reply setup.cfg:7]: Not sure if anyone manually runs just `pytest`,  but would not having `norecursedirs` make things a lot slower for them?

**Review by wedamija — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -6,3 +6,19 @@ build-backend = "setuptools.build_meta"
...
+addopts = "-ra --tb=short --strict-markers -p no:celery -p no:warnings"
+# TODO: --import-mode=importlib will become the default soon,
+# currently we have a few relative imports that don't work with that.
+markers = [
+    "snuba: mark a test as requiring snuba",
+]
+selenium_driver = "chrome"
+# This is for people who install pytest-xdist locally,
+# and use the -f/--looponfail feature.
+looponfailroots = ["src", "tests"]
```
Comment (billyvg) [inline pyproject.toml:24]: ty

**Review by billyvg — APPROVED** [review]

**Review by armenzg — COMMENTED** [review]
Thank you for tackling this!

**Pull request merged**